snapshot/git/repo/repo.go

```diff
@@ -0,0 +1,72 @@
+package repo
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+	"strings"
+)
+
+// Utilities for operating on a git repo.
+// Scoot often ends up with multiple git repos. E.g., one reference repo and then each
+// checkout is in its own repo.
+
+// Repository represents a valid Git repository.
+type Repository struct {
+	dir string
+}
+
+// Where r lives on disk
+func (r *Repository) Dir() string {
+	return r.dir
+}
+
+// Run a git command in r
+func (r *Repository) Run(args ...string) (string, error) {
+	log.Println("repo.Repository.Run", args)
+	cmd := exec.Command("git", args...)
+	cmd.Dir = r.dir
+	data, err := cmd.Output()
+	log.Println("repo.Repository.Run complete", err)
+	return string(data), err
+}
+
+// Run a git command that returns a sha.
+func (r *Repository) RunSha(args ...string) (string, error) {
+	out, err := r.Run(args...)
+	if err != nil {
+		return out, err
+	}
+	return validateSha(out)
+}
+
+// Checkout an id in r
+func (r *Repository) Checkout(id string) error {
+	_, err := r.Run("checkout", id)
+	return err
+}
+
+func validateSha(sha string) (string, error) {
```

Comment:
Depending where sha could end up coming from, it might make sense to accept 41 w/ '\n' or 40 w/out